plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1656,8 +1678,9 @@ bool producer_plugin_impl::remove_expired_trxs( const fc::time_point& deadline )
    size_t num_expired_other = 0;
    size_t orig_count = _unapplied_transactions.size();
    bool exhausted = !_unapplied_transactions.clear_expired( pending_block_time, deadline,
-                  [&num_expired_persistent, &num_expired_other, pbm = _pending_block_mode,
+                  [this, &num_expired_persistent, &num_expired_other, pbm = _pending_block_mode,
```

Comment:
Instead of capturing `this`, how about just capturing `&chain_plug`

Reply:
Yes, that would be much better.

Reply:
Got error `error: 'chain_plug' in capture list does not name a variable` after replacing `this` with `&chain_plug`. Did I miss somthing?

Reply:
`[chain_plug = chain_plug, ..]` should work.

Reply:
Yes. It works. Why didn't `&chain_plug` work here?

Reply:
Fixed.

Reply:
"Class members cannot be captured explicitly by a capture without initializer" https://en.cppreference.com/w/cpp/language/lambda

Reply:
Thanks!